### PR TITLE
Fix throwing exception when not local user

### PR DESCRIPTION
**Pull request by hwupathum**
Fix the bug introduced by https://github.com/wso2-extensions/identity-outbound-auth-email-otp/pull/164 where an exception can be thrown if isLocaluser() is false

### CODE REVIEW

**Review by ashensw — DISMISSED** [review]

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java [hunk 1]
```diff
@@ -1045,11 +1045,7 @@ protected void processEmailOTPFlow(HttpServletRequest request, HttpServletRespon
                 Boolean.parseBoolean(emailOTPParameters.get(EmailOTPAuthenticatorConstants.SHOW_AUTH_FAILURE_REASON));
         AuthenticatedUser authenticatedUser = getAuthenticatedUser(context);
         try {
-            if (authenticatedUser == null) {
-                throw new AuthenticationFailedException("Authentication failed!. Cannot proceed further without " +
-                        "identifying the user");
-            }
-            if (isLocalUser(context) && EmailOTPUtils.isAccountLocked(authenticatedUser)) {
+            if (isLocalUser(context) && authenticatedUser != null && EmailOTPUtils.isAccountLocked(authenticatedUser)) {
```
Comment (asha15) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: what if the authenticated user is null ? How are we handling that scenario ?

**Comment by hwupathum** [comment]
Closing the PR to handle authenticating user is null scenario.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java (hunk 1 above)
Comment (hwupathum) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java:1048]: Updated the integration test to handle it https://github.com/wso2/product-is/pull/16609

Comment (hwupathum) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: The authenticated user can be null if the subject attribute is not enabled, hence the null check cannot be used

Comment (hwupathum) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java]: authenticatedUser cannot be null if isLocalUser() is ture. therfore no need of a null check here

**Review by ashendes — APPROVED** [review]

**Pull request merged**